packages/inspection-report/src/components/Gallery.js

```diff
@@ -0,0 +1,180 @@
+import React, { useCallback, useState } from 'react';
+import { Modal, StyleSheet, View, Image, Pressable, Text } from 'react-native';
+import { MaterialIcons } from '@expo/vector-icons';
+
+import Thumbnail from './common/Thumbnail';
+
+const styles = StyleSheet.create({
+  container: {
+    alignContent: 'flex-start',
+    flex: 1,
+    flexDirection: 'row',
+    flexWrap: 'wrap',
+    gap: 10,
+    justifyContent: 'center',
+    paddingVertical: 15,
+  },
+  closeButtonWrapper: {
+    display: 'flex',
+    flexDirection: 'row',
+    justifyContent: 'space-between',
+    color: '#fff',
+    alignItems: 'center',
+    paddingHorizontal: 15,
+    paddingVertical: 5,
+    backgroundColor: '#121212',
+  },
+  header: {
+    fontSize: 20,
+  },
+});
+
+const pictureList = [
+  {
+    url: 'https://images.unsplash.com/photo-1551782450-a2132b4ba21d',
+    label: {
+      fr: 'Hood',
+      en: 'Hood',
+    },
+  },
+  {
+    url: '',
+    label: {
+      fr: 'Burger',
+      en: 'Burger',
+    },
+  },
+  {
+    url: '',
+    label: {
+      fr: 'Front Full Left',
+      en: 'Front Full Left',
+    },
+  },
+  {
+    url: '',
+    label: {
+      fr: 'Coffee',
+      en: 'Coffee',
+    },
+  },
+  {
+    url: '',
+    label: {
+      fr: 'Front Lateral Left',
+      en: 'Front Lateral Left',
+    },
+  },
+  {
+    url: '',
+    label: {
+      fr: 'Rocker Panel Left',
+      en: 'Rocker Panel Left',
+    },
+  },
+  {
+    url: '',
+    label: {
+      fr: 'Basketball',
+      en: 'Basketball',
+    },
+  },
+  {
+    url: '',
+    label: {
+      fr: 'Fern',
+      en: 'Fern',
+    },
+  },
+  {
+    url: '',
+    label: {
+      fr: 'Mushrooms',
+      en: 'Mushrooms',
+    },
+  },
+  {
+    url: '',
+    label: {
+      fr: 'Tomato basil',
+      en: 'Tomato basil',
+    },
+  },
+  {
+    url: '',
+    label: {
+      fr: 'Sea star',
+      en: 'Sea star',
+    },
+  },
+  {
+    url: '',
+    label: {
+      fr: 'Bike',
+      en: 'Bike',
+    },
+  },
+];
+
+export default function Gallery() {
+  const [photo, setPhoto] = useState('');
+  const [visibleImageInFullScreen, setVisibleImageInFullScreen] = useState(false);
```

Comment:
no need for this variable. You can know when the modal must be displayed based on if the photo variable is `null` or not